src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerModifierAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Immutable;
+using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PrivateEventHandlers
+{
+	public class EventHandlerModifierAnalyzer : PXGraphAggregatedAnalyzerBase
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphEventSemanticModel graphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			foreach (var handler in graphExtension.GetAllEvents().Where(graphEvent => graphEvent.Symbol.IsDeclaredInType(graphExtension.Symbol)))
```

Comment:
Please avoid using big LINQ queries directly in `foreach` loops. This makes the debug more difficult since it's harder to see what items will be collected for the iteration. A separate variables prevents such problems:
```cs
var declaredEventHandlers = 
        graphExtension.GetAllEvents()
                               .Where(graphEvent => graphEvent.Symbol.IsDeclaredInType(graphExtension.Symbol));
foreach (var handler in declaredEventHandlers)
{
   ...
}
```

Reply:
Done.